lib/fog/aws/credential_fetcher.rb

```diff
@@ -21,7 +21,7 @@ def fetch_credentials(options)
           if options[:use_iam_profile]
             begin
               role_data = nil
-              region = options[:region] || ENV["AWS_DEFAULT_REGION"]
+              region = options[:region] || ENV["AWS_REGION"] || ENV["AWS_DEFAULT_REGION"]
```

Comment:
Actually, I question whether the environment is ever used because `options[:region]` appears to default to `us-east-1` in https://github.com/fog/fog-aws/blob/b96b27a6bc5d8ada0672abd98a2de330c99b3426/lib/fog/aws/storage.rb#L298-L304?

Reply:
Never mind, I confirmed that this works properly if you use `Fog::Storage.new` since fog-core appears to call `fetch_credentials` directly from `handle_settings`.